apps/openmw/mwmechanics/actors.cpp

```diff
@@ -306,7 +306,25 @@ namespace MWMechanics
         // Get actors allied with actor1. Includes those following or escorting actor1, actors following or escorting those actors, (recursive)
         // and any actor currently being followed or escorted by actor1
         std::set<MWWorld::Ptr> allies1;
-        getActorsSidingWith(actor1, allies1);
+
+        // If we have already found actor1's allies, use the cache
+        auto search = cachedAllies.find(actor1);
+        if (search != cachedAllies.end())
+            allies1 = search->second;
+        else
+        {
+            getActorsSidingWith(actor1, allies1);
+
+            // Cache ptrs and their sets of allies so we don't search for them again the next time engageCombat is called
+            cachedAllies.insert(std::make_pair(actor1, allies1));
+
+            for (std::set<MWWorld::Ptr>::const_iterator it = allies1.begin(); it != allies1.end(); ++it)
+            {
+                search = cachedAllies.find(*it);
+                if (search == cachedAllies.end())
+                    cachedAllies.insert(std::make_pair(*it, allies1));
+            }
+        }
```

Comment:
* Don't use auto (C++11)
* I see this code repeated 3 times. Probably best to move it into a function.